src/core/interfaces.ts

```diff
@@ -46,4 +46,12 @@ module Plottable {
     topLeft: Point;
     bottomRight: Point;
   };
+
+  export interface Entity<C extends Component> {
+    datum: any;
+    index: number;
+    position: Point;
+    selection: d3.Selection<any>;
+    component: C;
```

Comment:
The type of `Position` is `Point`, so that should be enough indication for Typescript consumers. I did consider the name `Location` though.

`index` doesn't make much sense without `Dataset`, so I think I'll move that to `PlotEntity`.